### PR TITLE
feat: Apply latest 1.2 working draft

### PR DESCRIPTION
**Pull request by eps1lon**
- [x] new roles
- [x] changes to roles 
  - [x] abstract roles
  - [x] widget roles
  - [x] composite widget roles
  - [x] document structure
  - [x] landmark roles
  - [x] live region roles
  - [x] window roles
- ~[ ] concept changes~
  Reverted the concept changes. Currently aria-query uses these to apply constraints from [html-aam](https://www.w3.org/TR/html-aam-1.0/). To avoid future confusion we should include the mappings separately (and get rid of related concepts if that isn't used for anything else).

The changelog of the spec is unusable for this work. It's sorted by time and misses almost all the changes we're interested in for this package.

For reviewers:
- Review on a per-commit basis. Changes were auto-generated so I had to include other changes such as sorting or formatting. Non-substantive changes are contained in separate commits.
- https://www.w3.org/TR/wai-aria-1.2/#roles_categorization
- https://rawgit.com/w3c/aria/master/#role_definitions

## Test plan
- [x] CI green
- [x] integrates (using codesandbox deploy from https://github.com/eps1lon/aria-query/pull/4) with
  - [x] `@testing-library/dom`: https://github.com/eps1lon/dom-testing-library/tree/feat/aria-12 (failures are due to new roles)
  - [x] `eslint-plugin-jsx-a11y`: Needs some work. I can help a bit since I've debugged one issue where we have conditions such as `!isInteractiveElement && !isNonInteractiveElement`. Without knowing implementation this expression is impossible to begin with. These fail now because a `div` is considered to be a NonInteractiveElement because of the `generic` role. For some reason `interactive-supports-focus` decides to not create an error if it encounters these.

### CODE REVIEW

**Comment by jessebeach** [comment]
I'm pretty excited about this! I need to bring this into the FB codebase and run a side-by-side comparison running out lint rules to determine the delta between the two version. On the surface, the changes make sense. I'll try to do this tomorrow.


Thanks @eps1lon

**Comment by eps1lon** [comment]
@jessebeach https://github.com/evcohen/eslint-plugin-jsx-a11y/compare/master...eps1lon:feat/aria-12 includes fixes to `interactive-supports-focus` (implementation fix) and `role-supports-aria-props` (test-only fix).

**Comment by jessebeach** [comment]
@eps1lon I had used the term "static" to refer to elements that previously had no semantic mapping -- neither interactive nor non-interactive. It appears now that these elements are given the `generic` role. But they are still neither interactive nor non-interactive.

**Comment by jessebeach** [comment]
I ran a side by side test of the eslint-plugin-jsx-a11y plugin and found 705 more issues over the same set of files (~150,000). So there's definitely some work to do on that plugin to accommodate these updates. I'm looking into it today.

**Comment by jessebeach** [comment]
@eps1lon one point of process that will help in the future for large changes like this. You've mixed substantive changes (spec upgrades) with non-substantive changes (alphabetizing the props in roles.json). This makes it difficult to determine causes of test failures. You could instead put up a PR to do the housekeeping work, then another PR to effect the behavior changes.

**Comment by eps1lon** [comment]
> This makes it difficult to determine causes of test failures. You could instead put up a PR to do the housekeeping work, then another PR to effect the behavior changes.

Absolutely agree. Though I deliberately put these changes into separate commits so you can still review substantive and non-substantive changes individually by clicking through individual commits:  [https://github.com/A11yance/aria-query/pull/46/commits/3c2f213abf61c9523493fbd808ad0178f23288ee ](https://github.com/A11yance/aria-query/pull/46/commits/3c2f213abf61c9523493fbd808ad0178f23288ee)

**Comment by jessebeach** [comment]
> > This makes it difficult to determine causes of test failures. You could instead put up a PR to do the housekeeping work, then another PR to effect the behavior changes.
> 
> Absolutely agree. Though I deliberately put these changes into separate commits so you can still review substantive and non-substantive changes individually by clicking through individual commits: [https://github.com/A11yance/aria-query/pull/46/commits/3c2f213abf61c9523493fbd808ad0178f23288ee ](https://github.com/A11yance/aria-query/pull/46/commits/3c2f213abf61c9523493fbd808ad0178f23288ee)

That's true. Github's merging doesn't let me cherrypick, though. The workflow encourages atomic 
PRs. If you have time to PR the first two commits in the stack, please do and I'll get those chore commits merged today. I'll continue to debug on the eslint-plugin-jsx-a11y side.

File: scripts/roles.json [hunk 1]
```diff
@@ -4430,116 +5237,119 @@
...
     ],
-    "childrenPresentational": true
-  },
-  "table": {
-    "superClass": ["section"],
-    "accessibleNameRequired": true,
-    "nameFrom": ["author"],
-    "requiredContextRole": [],
-    "requiredOwnedElements": [
-      ["row"],
```
Comment (jessebeach) [inline scripts/roles.json:4529]: There are several examples like this change where arrays are removed, creating mixed content inside `requiredOwnedElements`. Could you revert these changes so that `requiredOwnedElements` consistently contains arrays?

File: scripts/roles.json [hunk 2]
```diff
@@ -2463,1293 +2686,1377 @@
...
-    "childrenPresentational": true
-  },
-  "input": {
-    "superClass": ["widget"],
-    "accessibleNameRequired": false,
-    "nameFrom": ["author"],
     "requiredContextRole": [],
     "requiredOwnedElements": [],
+    "requiredProps": [],
+    "superClass": ["landmark"]
```
[the remark this answers is not in the document]
Comment (eps1lon) [reply scripts/roles.json]: done

**Comment by eps1lon** [comment]
Fixed https://github.com/A11yance/aria-query/pull/46#discussion_r439229900 by updating my parser script which revealed a bunch of minor updates that weren't included. Should now be up-to-date with ARIA 1.2 WD 18 December 2019

File: scripts/roles.json (hunk 1 above)
Comment (eps1lon) [reply scripts/roles.json:4529]: Done

**Comment by jessebeach** [comment]
> For some reason interactive-supports-focus decides to not create an error if it encounters these.

@eps1lon good catch. We're probably missing test coverage here. I'll look into it today.
